ui/storybook/README.md

```diff
@@ -0,0 +1,122 @@
+# Storybook
+
+This package is used to generate documentation for Perses UI components using [Storybook](https://storybook.js.org/).
+
+> This project currently uses the [v7 beta](https://storybook.js.org/docs/7.0/react/get-started/introduction) of Storybook. Be sure to reference the associated [v7 beta documentation](https://storybook.js.org/docs/7.0/react/get-started/introduction).
+
+## Getting started
+
+### Running storybook
+
+- Run storybook in development mode: `npm run storybook`
+
+### Building storybook
+
+- Build only: `npm run storybook:build`
+- Build & serve static assets from build: `npm run storybook:serve`
+
+## Creating stories
+
+### Where should stories live?
+
+Stories **SHOULD** live in the first entry in the list below that matches their purpose:
+
+- Stories that are tightly coupled with a component should be colocated with the component code in the associated package (e.g. the stories for the `LineChart` component live at `ui/components/src/LineChart/LineChart.stories.tsx` as a sibling of `LineChart.tsx`).
+- Package-specific stories should live in the `src/stories` directory for that package.
+- Stories that provide project-wide documentation or involve components from multiple packages should live in the storybook package (`ui/storybook/stories`).
+
+### What kind of story format should I use?
+
+#### Components
+
+Stories for components should use [Component Story Format (CSF)](https://storybook.js.org/docs/react/api/csf) and follow the `PascalCaseName.stories.tsx` naming scheme.
+
+See [Writing stories in TypeScript](https://storybook.js.org/blog/writing-stories-in-typescript/) for some additional guidance on typing for stories.
+
+#### Text-only documentation
+
+Stories that are primarily text-based documentation (e.g. the `About` pages for each package) should be written using [MDX](https://storybook.js.org/docs/react/writing-docs/mdx) and follow the `PascalCase.stories.mdx` naming scheme.
+
+These may include importing a few small components like the `LinkTo` component from the [Link addon](https://storybook.js.org/addons/@storybook/addon-links/) to link out to other stories.
+
+### Importing package code
+
+#### TLDR;
+
+Stories should import package code by referencing the package, not by making a relative reference. Under the hood, this will reference source code (not compiled code), so we can benefit from some Storybook features like autogenerating documentation from types.
+
+Import like this in a story:
+
+```ts
+// Import from the package. Storybook's webpack config will alias this to the
+// associated source code.
+import { LineChart } from '@perses-dev/components';
+```
+
+Not like this:
+
+```ts
+// Importing directly from the source file will make it easier to forget to
+// export components for consumption from the main export for the package.
+import { LineChart } from '.';
+```
+
+#### Long version
```

Comment:
This section describes one of the most controversial things I'm doing in this setup. Please review and let me know if you really dislike these tradeoffs and have suggestions for other approaches.

Reply:
I forget what the downside to doing it this way are (just that we're not testing our compiled code?)

Reply:
Is it also sort of more abnormal / not the usual way to do it? (it's more common to import compiled code)?

Reply:
Primarily that we're not testing our compiled code. There's also a bit of complexity in the webpack/ts config to do the aliasing. Because of this limitation of storybook, most people who write typescript point at their source code. I ran into a few discussions/issues/etc. asking about supporting compiled code with `.d.ts` files from a while back and there was little-to-no response, so I don't think there's a ton of appetite to support it.